plugins/builds/index.js

```diff
@@ -528,27 +528,29 @@ async function getParentBuildStatus({ newBuild, joinListNames, pipelineId, build
  * @return {Promise}                The newly updated/created build
  */
 async function handleNewBuild({ done, hasFailure, newBuild, jobName, pipelineId }) {
-    if (done) {
-        // Delete new build since previous build failed
-        if (hasFailure) {
-            logger.info(
-                `Failure occurred in upstream job, removing new build - build:${newBuild.id} pipeline:${pipelineId}-${jobName} event:${newBuild.eventId} `
-            );
-            await newBuild.remove();
-
-            return null;
-        }
+    if (!done) {
+        return null;
+    }
+    if (!['CREATED', null, undefined].includes(newBuild.status)) {
+        return null;
+    }
 
-        // If all join builds finished successfully and it's clear that a new build has not been started before, start new build
-        if (['CREATED', null, undefined].includes(newBuild.status)) {
-            newBuild.status = 'QUEUED';
-            const queuedBuild = await newBuild.update();
+    // Delete new build since previous build failed
+    if (hasFailure) {
+        logger.info(
+            `Failure occurred in upstream job, removing new build - build:${newBuild.id} pipeline:${pipelineId}-${jobName} event:${newBuild.eventId} `
+        );
+        await newBuild.remove();
 
-            return queuedBuild.start();
-        }
+        return null;
     }
 
-    return null;
+    // All join builds finished successfully and it's clear that a new build has not been started before.
+    // Start new build.
+    newBuild.status = 'QUEUED';
+    await newBuild.update();
+
+    return newBuild.start();
```

Comment:
Elsewhere, the return value of `update` is used. Wouldn't it be better to follow that?

Reply:
The update function just returns the instance itself, so there seems no need to reassign the return value and use it.

In addition, I considered that it would be more dangerous to cause the misunderstanding that it was returning a copy of the instance by returning.

Reply:
OK.